src/app/lightning/lookup-invoice-by-hash.ts

```diff
@@ -0,0 +1,21 @@
+import { LndService } from "@services/lnd"
+import { offchainLnds } from "@services/lnd/utils"
+import { InvoiceNotFoundError } from "@domain/bitcoin/lightning"
+
+export const lookupInvoiceByHash = async (
+  paymentHash: PaymentHash,
+): Promise<LnInvoiceLookup | ApplicationError> => {
+  const lndService = LndService()
+  if (lndService instanceof Error) return lndService
+
+  for (const { pubkey } of offchainLnds) {
```

Comment:
This should not be exposed here. That the `LndService` is doing client-side load balancing is an implementation detail. The loop should be hidden from the caller.